client/starwhale/api/_impl/data_store.py

```diff
@@ -1120,13 +1130,337 @@ def get_compressor(file: Path) -> Compressor:
+            if start is not None and key < start:
+                continue
+
+            if end_inclusive:
+                if end is not None and key > end:
+                    break
+            else:
+                if end is not None and key >= end:
+                    break
+
+            yield record
+
+    def get(self, key: Any) -> Optional[InnerRecord]:
+        return self.records.get(key, None)  # type: ignore
+
+    def _must_valid_key_type(self, key: Any) -> None:
+        key_type = _get_type(key)
+        if self._key_type is None:
+            self._key_type = key_type
+        elif self._key_type != key_type:
+            raise RuntimeError(
+                f"conflicting key type, expected {self._key_type}, actual {key_type}"
+            )
+
+    def insert_record(self, record: InnerRecord) -> None:
+        self.dirty = True
+        key = record.key
+        self._must_valid_key_type(key)
+        ir = self.records.get(key, None)
+        if ir is None:
+            self.records[key] = record
+        else:
+            ir.update(record)
+
+    def insert_dict(self, record: Dict[str, Any]) -> str:
+        self.dirty = True
+        key = record[self.key_column]
+        self._must_valid_key_type(key)
+        ir = self.get(key)
+        if ir is None:
+            ir = InnerRecord(self.key_column)
+            self.records[key] = ir
+        return ir.append(Record(copy.deepcopy(record)))
+
+    def delete(self, keys: List[Union[int, str]]) -> str:
+        """
+        Delete records by keys. If the key is not found, it will be ignored.
+        Returns the sequence number of the last delete operation, or None if no delete operation is performed.
+        """
+        if len(keys) == 0:
+            raise RuntimeError("keys can not be empty")
+
+        revision = None
+        self.dirty = True
+        for key in keys:
+            r = self.get(key)
+            if r is None:
+                r = InnerRecord(self.key_column)
+                self.records[key] = r
+
+            del_mark = Record({self.key_column: key, "-": True})
+            revision = r.append(del_mark)
+        return revision  # type: ignore
+
+
+class DataBlock:
+    def __init__(
+        self,
+        min_key: Any,
+        max_key: Any,
+        key_column: str,
+        block_id: int | None = None,
+        file: Path | None = None,
+    ) -> None:
+        self.min_key = min_key
+        self.max_key = max_key
+        self.key_column = key_column
+        self.file = file
+        self.block_id = block_id
+
+    @property
+    def virtual(self) -> bool:
+        return self.file is None
+
+    def ahead(self, key: Any) -> bool:
+        return key is not None and self.max_key < key
+
+    def behind(self, key: Any) -> bool:
+        return key is not None and self.min_key > key
+
+    def __ge__(self, other: object) -> Any:
+        if not isinstance(other, DataBlock):
+            raise NotImplementedError
```

Comment:
`return NotImplementedError`?

Reply:
All changed to `raise NotImplementedError`